dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -2193,4 +2284,62 @@ public File generateJson(File targetJsonDir, List<File> esas) throws IOException
         return result;
     }
 
+    private List<File> downloadPublicKeys() {
+        VerifySignatureUtility verifyUtility = new VerifySignatureUtility();
+        VerifyOption verify = (VerifyOption) data.get(InstallConstants.VERIFY_OPTION);
+        Collection<Map<String, String>> usrkeys = (Collection<Map<String, String>>) data.get(InstallConstants.USER_PUBLIC_KEYS);
+        try {
+            pubKeys = verifyUtility.downloadPublicKeys(usrkeys, verify, envMap);
+        } catch (InstallException e) {
+            data.put(InstallConstants.ACTION_RESULT, ERROR);
+            data.put(InstallConstants.ACTION_ERROR_MESSAGE, e.getMessage());
+            data.put(InstallConstants.ACTION_EXCEPTION_STACKTRACE, ExceptionUtils.stacktraceToString(e));
```

Comment:
would we normally log the exception here as well or does this `data` object get logged?